### PR TITLE
Inline gpurt traceRay entryfunction

### PR DESCRIPTION
**Pull request by jiaolu**
This is for future refactoring where payload or callable data can refactored directly in the lowerRaytracing class

### CODE REVIEW

**Review by LLJJDD — COMMENTED** [review]
Looks good, but do we really need this? I think it won't hurt to process something later based on the assumption that LowerGpurt pass will run after SpirvLowerRayTracing pass.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 4983e02926d7b23755c85a36266a23a44c2de00a
<details><summary>CTS tests (Failed: 0/211876)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35146/70631 (49.8%)</li>
<li>Failed: 0/70631 (0.0%)</li>
<li>Not Supported: 35485/70631 (50.2%)</li><li>Warnings: 0/70631 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39023/70629 (55.3%)</li>
<li>Failed: 0/70629 (0.0%)</li>
<li>Not Supported: 31606/70629 (44.7%)</li><li>Warnings: 0/70629 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39057/70616 (55.3%)</li>
<li>Failed: 0/70616 (0.0%)</li>
<li>Not Supported: 31559/70616 (44.7%)</li><li>Warnings: 0/70616 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by jiaolu** [comment]
> Looks good, but do we really need this? I think it won't hurt to process something later based on the assumption that LowerGpurt pass will run after SpirvLowerRayTracing pass.

yeah, it is a little subjective, The payload is more raytracing to me, so its processing probably stays at SpirvlowerRaytracing while LowerGPURT is stuff more related to its own library processing itself which can be shared beyond raytracing probably.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 76f4433420fe84007f85b4e76b4984a0c67038cd
<details><summary>CTS tests (Failed: 0/211879)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 8.9, Gfx10</summary><ul>

<li>Passed: 35147/70631 (49.8%)</li>
<li>Failed: 0/70631 (0.0%)</li>
<li>Not Supported: 35484/70631 (50.2%)</li><li>Warnings: 0/70631 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39025/70632 (55.3%)</li>
<li>Failed: 0/70632 (0.0%)</li>
<li>Not Supported: 31607/70632 (44.7%)</li><li>Warnings: 0/70632 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39057/70616 (55.3%)</li>
<li>Failed: 0/70616 (0.0%)</li>
<li>Not Supported: 31559/70616 (44.7%)</li><li>Warnings: 0/70616 (0.0%)</li>
</ul></details>


</ul></details>

File: llpc/lower/llpcSpirvLowerRayTracing.cpp [hunk 1]
```diff
@@ -494,15 +497,16 @@ void SpirvLowerRayTracing::visitReportHitOp(ReportHitOp &inst) {
 // @param [in/out] module : LLVM module to be run on
 // @param [in/out] analysisManager : Analysis manager to use for this transformation
 PreservedAnalyses SpirvLowerRayTracing::run(Module &module, ModuleAnalysisManager &analysisManager) {
-  runImpl(module);
+  runImpl(module, analysisManager);
```
Comment (nhaehnle) [inline llpc/lower/llpcSpirvLowerRayTracing.cpp:500]: We could just merge run and runImpl entirely. The split isn't needed anymore.

**Comment by amdvlk-admin** [comment]
### Test summary for commit b5d26f6603b9bf5155d627cc46af152ebec55f96
<details><summary>CTS tests (Failed: 0/211876)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35146/70631 (49.8%)</li>
<li>Failed: 0/70631 (0.0%)</li>
<li>Not Supported: 35485/70631 (50.2%)</li><li>Warnings: 0/70631 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39023/70629 (55.3%)</li>
<li>Failed: 0/70629 (0.0%)</li>
<li>Not Supported: 31606/70629 (44.7%)</li><li>Warnings: 0/70629 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39057/70616 (55.3%)</li>
<li>Failed: 0/70616 (0.0%)</li>
<li>Not Supported: 31559/70616 (44.7%)</li><li>Warnings: 0/70616 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — COMMENTED** [review]
I think this makes sense, but I've got to ask: Instead of first creating the separate function and then inlining it, couldn't we just insert the code into the caller directly?

**Comment by jiaolu** [comment]
> I think this makes sense, but I've got to ask: Instead of first creating the separate function and then inlining it, couldn't we just insert the code into the caller directly?

Do you mean insert code of traceRay to the _cs_ function? traceRay fucntion is not fixed , it is configured from gpurtshim, i think it might easier to make call and inline.

**Comment by jiaolu** [comment]
retest this please

**Comment by jiaolu** [comment]
retest this please

**Comment by jiaolu** [comment]
retest this please

**Comment by nhaehnle** [comment]
> > I think this makes sense, but I've got to ask: Instead of first creating the separate function and then inlining it, couldn't we just insert the code into the caller directly?
> 
> Do you mean insert code of traceRay to the _cs_ function? traceRay fucntion is not fixed , it is configured from gpurtshim, i think it might easier to make call and inline.

Ah yes, I understand this now. Sorry for the confusion.

File: llpc/lower/llpcSpirvLowerRayTracing.cpp [hunk 2]
```diff
@@ -512,15 +506,15 @@ bool SpirvLowerRayTracing::runImpl(Module &module) {
...
   initGlobalCallableData();
-
+  Instruction *insertPos = nullptr;
   // Create empty raygen main module
   if (module.empty()) {
     m_shaderStage = ShaderStageRayTracingRayGen;
     createRayGenEntryFunc();
     rayTracingContext->setEntryName("main");
-    return true;
+    goto end;
```
Comment (nhaehnle) [inline llpc/lower/llpcSpirvLowerRayTracing.cpp:515]: Please avoid `goto`, here you can just return directly.

**Review by nhaehnle — APPROVED** [review]
Thanks!

**Comment by jiaolu** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 2d4337627d49ae60cadc53f5611113d79e4ec107
<details><summary>CTS tests (Failed: 0/211876)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35146/70631 (49.8%)</li>
<li>Failed: 0/70631 (0.0%)</li>
<li>Not Supported: 35485/70631 (50.2%)</li><li>Warnings: 0/70631 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39023/70629 (55.3%)</li>
<li>Failed: 0/70629 (0.0%)</li>
<li>Not Supported: 31606/70629 (44.7%)</li><li>Warnings: 0/70629 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39057/70616 (55.3%)</li>
<li>Failed: 0/70616 (0.0%)</li>
<li>Not Supported: 31559/70616 (44.7%)</li><li>Warnings: 0/70616 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**